app/controllers/spaceship_controller.rb

```diff
@@ -0,0 +1,32 @@
+class SpapceshipsController < ApplicationController
+	before_action :authenticate_user!
+
+    def index
+      @spaceships = Spaceship.all
+    end
+
+    def new
+      @spaceship = Spaceship.new
+    end
+
+    def create
+			@spaceship = Spaceship.new
+			if @spaceship.save
+				redirect_to spaceship_path(@spaceship)
+			else
+				render 'new'
+		end
+	end
+
+		def show
+		@spaceship = Spaceship.show
+		end
```

Comment:
Check indentation please. It's quite off.

Reply:
mmm yeh could be better